File: java/src/main/java/problems1_100/PlusOne.java
package problems1_100;

enum PlusOne {
    DEFAULT {
        @Override
        int[] plusOne(int[] digits) {
            int carry = 0;
            for (int i = digits.length - 1; i >= 0; i--) {
                if (i == digits.length - 1) {
                    digits[i] = digits[i] + 1;
                }
                digits[i] = digits[i] + carry;
                carry = digits[i] / 10;
                digits[i] = digits[i] % 10;
            }
            if (carry == 1) {
                int[] digits2 = new int[digits.length + 1];
                digits2[0] = 1;
                System.arraycopy(digits, 0, digits2, 1, digits2.length - 1);
                return digits2;
            }
            return digits;
        }
    };

    abstract int[] plusOne(int[] digits);
}

File: kotlin/src/main/kotlin/problems1_100/PlusOne.kt
package problems1_100

enum class PlusOne {
    DEFAULT {
        override fun plusOne(digits: IntArray): IntArray {
            var carry = 0
            for (i in digits.indices.reversed()) {
                if (i == digits.size - 1) {
                    digits[i] = digits[i] + 1
                }
                digits[i] = digits[i] + carry
                carry = digits[i] / 10
                digits[i] = digits[i] % 10
            }
            if (carry == 1) {
                val digits2 = IntArray(digits.size + 1)
                digits2[0] = 1
                System.arraycopy(digits, 0, digits2, 1, digits2.size - 1)
                return digits2
            }
            return digits
        }
    },
    ;

    abstract fun plusOne(digits: IntArray): IntArray
}
